src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/web/WorkflowRunController.kt

```diff
@@ -1,37 +1,117 @@
 package org.vaccineimpact.orderlyweb.controllers.web
 
+import com.github.fge.jackson.JsonLoader
+import com.github.fge.jsonschema.main.JsonSchemaFactory
+import com.google.gson.annotations.SerializedName
 import org.vaccineimpact.orderlyweb.ActionContext
+import org.vaccineimpact.orderlyweb.OrderlyServer
+import org.vaccineimpact.orderlyweb.OrderlyServerAPI
+import org.vaccineimpact.orderlyweb.Serializer
 import org.vaccineimpact.orderlyweb.controllers.Controller
+import org.vaccineimpact.orderlyweb.db.AppConfig
 import org.vaccineimpact.orderlyweb.db.repositories.OrderlyWebWorkflowRunRepository
 import org.vaccineimpact.orderlyweb.db.repositories.WorkflowRunRepository
+import org.vaccineimpact.orderlyweb.errors.BadRequest
+import org.vaccineimpact.orderlyweb.models.WorkflowRun
+import org.vaccineimpact.orderlyweb.models.WorkflowRunRequest
 import org.vaccineimpact.orderlyweb.models.WorkflowRunSummary
 import org.vaccineimpact.orderlyweb.viewmodels.WorkflowRunViewModel
-import org.vaccineimpact.orderlyweb.models.WorkflowRun
+import java.io.File
+import java.net.HttpURLConnection.HTTP_OK
+import java.time.Instant
 
 class WorkflowRunController(
     context: ActionContext,
-    private val workflowRunRepository: WorkflowRunRepository
+    private val workflowRunRepository: WorkflowRunRepository,
+    private val orderlyServerAPI: OrderlyServerAPI
 ) : Controller(context)
 {
-    constructor(context: ActionContext) : this(context, OrderlyWebWorkflowRunRepository())
-
-    fun getWorkflowRunDetails(): WorkflowRun
-    {
-        val key = context.params(":key")
-        return workflowRunRepository.getWorkflowRunDetails(key)
-    }
+    constructor(context: ActionContext) : this(
+        context,
+        OrderlyWebWorkflowRunRepository(),
+        OrderlyServer(AppConfig()).throwOnError()
+    )
 
     @Template("run-workflow-page.ftl")
     fun getRunWorkflow(): WorkflowRunViewModel
     {
         return WorkflowRunViewModel(context)
     }
 
+    fun getWorkflowRunDetails(): WorkflowRun
+    {
+        val key = context.params(":key")
+        return workflowRunRepository.getWorkflowRunDetails(key)
+    }
+
     fun getWorkflowRunSummaries(): List<WorkflowRunSummary>
     {
         return workflowRunRepository.getWorkflowRunSummaries(
             context.queryParams("email"),
             context.queryParams("namePrefix")
         )
     }
+
+    internal data class WorkflowRunResponse(
+        @SerializedName(value = "workflow_key")
+        val key: String,
+        val reports: List<String>
+    )
+
+    fun createWorkflowRun(): String
+    {
+        val workflowRunRequestJson = context.getRequestBody()
+
+        val isValidWorkflow = JsonSchemaFactory.byDefault()
+            .getJsonSchema(File("../../docs/spec/RunWorkflow.schema.json").toURI().toString())
```

Comment:
Just struck me that this might not work when the app is packaged as a jar - may need to load via classpath instead. Will check.